pkg/agent/schedwatch/watch.go

```diff
@@ -19,17 +21,54 @@ func isActivePod(pod *corev1.Pod) bool {
 	return pod.Status.PodIP != "" && util.PodReady(pod)
 }
 
+type SchedulerTracker struct {
+	current *currentSchedPod
```

Comment:
We don't directly change `current`, but we change the fields behind the pointer

Reply:
OK I see. It's tricky and  error-prone that `currentSchedPod`'s lock is managed by other entities, but I see why you needed to do it. Also, the fact that both `SchedulerTracker` and `schedPods` have the `current` field is not the best, since we have to ensure that these two are consistent but technically we can't always guarantee it.

A possible workaround to avoid mentioned problem is this:
```go
type SchedulerTracker struct {
	pods schedPods

	Stop func()
}

func (s SchedulerTracker) Get() *SchedulerInfo {
	return s.getCurrent()
}

type schedPods struct {
	mu  sync.RWMutex
	current *currentSchedPod
	pods    map[types.UID]*SchedulerInfo
}

func (s schedPods) getCurrent() {
        s.mu.RLock()
        defer s.mu.RUnlock()
        return s.current
}

type currentSchedPod struct {
	pod *SchedulerInfo
}
```